.github/tekton/pull-request.yaml

```diff
@@ -0,0 +1,148 @@
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: ks-pull-request-pipeline
+spec:
+  workspaces:
+    - name: repo
+  params:
+    - name: repo-full-name
+      description: "Repository full name. like: kubesphere-sigs/ks"
+    - name: clone-url
+      description: Git repository clone URL.
+    - name: revision
+      description: Git repository revision to checkout.
+    - name: dashboard-url
+      description: Tekton dashboard access URL.
+      default: "http://demo:31962/#/namespaces/ks/pipelineruns"
+  tasks:
+    - name: set-running-status
+      taskRef:
+        name: github-set-status
+      params:
+        - name: REPO_FULL_NAME
+          value: $(params.repo-full-name)
+        - name: SHA
+          value: $(params.revision)
+        - name: DESCRIPTION
+          value: Build has started
+        - name: STATE
+          value: pending
```

Comment:
I prefer to let it starts with a capital letter. For instance: `Pending`

Reply:
This value is from API documentation: https://docs.github.com/en/rest/reference/commits#commit-statuses. State with lower-case is needed here.

Reply:
Learned it.